pkg/stash/with_redis_sentinel.go

```diff
@@ -0,0 +1,37 @@
+package stash
+
+import (
+	errs "errors"
+
+	"github.com/go-redis/redis/v7"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/storage"
+)
+
+var (
+	// ErrNoEndpoints is returned when no endpoints were provided
+	ErrNoEndpoints = errs.New("no endpoints provided")
+)
+
+// WithRedisSentinelLock returns a distributed singleflight
+// with a redis cluster that utilizes sentinel for quorum and failover
+func WithRedisSentinelLock(endpoints []string, master, password string, checker storage.Checker) (Wrapper, error) {
```

Comment:
This could alternatively be compacted into `WithRedisLock`, but I didn't want to add _too_ much complexity. Thoughts?

Reply:
I like that you've separated sentinel from redis itself. sentinel is treated differently enough in the redis docs that I think we should treat it differently here from a docs and code perspective.